client/src/main/scala/org/http4s/client/PoolManager.scala

```diff
@@ -259,10 +279,17 @@ private final class PoolManager[F[_], A <: Connection[F]](
     }
   }
 
-  private def findFirstAllowedWaiter =
+  private def findFirstAllowedWaiter = {
+    val (expired, rest) = waitQueue.span(w => isExpired(w.at))
```

Comment:
Should this be partition? Do we want the longest prefix of the expired requests, or all of them?

Reply:
Great question.  The oldest should be at the front of the queue, so I think `span` is a fair optimization that prevents us from iterating the entire queue.

If this were a `takeWhile`, and we `drop`ped its size, we could get rid of the var.